K2Bridge/KustoDAL/KustoQueryExecutor.cs

```diff
@@ -57,13 +57,26 @@ internal class KustoQueryExecutor : IQueryExecutor
         /// <returns>A new instance of <see cref="KustoConnectionStringBuilder"/>.</returns>
         public static KustoConnectionStringBuilder CreateKustoConnectionStringBuilder(IConnectionDetails connectionDetails)
         {
-            var conn = new KustoConnectionStringBuilder(
-                connectionDetails.ClusterUrl,
-                connectionDetails.DefaultDatabaseName)
-                .WithAadApplicationKeyAuthentication(
-                    connectionDetails.AadClientId,
-                    connectionDetails.AadClientSecret,
-                    connectionDetails.AadTenantId);
+            KustoConnectionStringBuilder conn;
+
+            if (string.IsNullOrEmpty(connectionDetails.AadClientId) ||
```

Comment:
I'm not a big fan of this approach. If someone forgets to have a secret, rather than informing them we will silently do Managed Identity auth.

I think we should add an explicit option in the config for the type of the authentication, and then throw if keys are missing.

Reply:
Make sense.
I will introduce the `useManagedIdentity` bool variable in the **appsettings.json**.
The default value will be false to be backward compatible.